components/patient-data/ui/src/main/resources/PhenoTips/UIX_Field__genes.xml

```diff
@@ -120,105 +120,79 @@
-             var info = data &amp;&amp; data.response &amp;&amp; data.response[geneName] &amp;&amp; data.response[geneName].external_ids;
-             if (info) {
+  }
 
-                var geneLinks = element.next('div.gene-links') || new Element('div', {'class' : 'gene-links'});
-                var externalDBs = [
-                          { name : 'GENECARDS', url : 'http://www.genecards.org/cgi-bin/carddisp.pl?gene=',          field : 'genecards_id'},
-                          { name : 'OMIM',      url : 'http://www.omim.org/entry/',                                  field : 'omim_id'},
-                          { name : 'Entrez',    url : 'http://www.ncbi.nlm.nih.gov/gene/?term=',                     field : 'entrez_id'},
-                          { name : 'RefSeq',    url : 'http://www.ncbi.nlm.nih.gov/nuccore/',                        field : 'refseq_accession'},
-                          { name : 'Ensembl',   url : 'http://useast.ensembl.org/Homo_sapiens/Gene/Compara_Tree?g=', field : 'ensembl_gene_id'}
-                ];
-                externalDBs.each(function (item) {
-                         var value = info[item.field];
-                         if (value) {
-                           if (!value.each) {
-                             value = [value];
-                           }
-                           value.each (function (id) {
-                             geneLinks.insert(new Element('a', {
-                               'href'   : item.url + id,
-                               'class'  : 'button secondary'
-                             }).update(item.name + ': ' + id).wrap('span', {'class' : 'buttonwrapper'}));
-                           });
-                         }
-                });
-                geneLinks.select('a').each(function(item) {
-                         item.observe('click', function(event) {
-                            event.stop();
-                            window.open(item.href);
-                         })
-                });
-                if (!element.__handleEmptyValue) {
-                   element.__handleEmptyValue = true;
-                   ['keyup', 'change', 'keypress'].each(function(eventName) {
-                     element.observe(eventName, function (event) {
-                       if (element.value == '') {
-                         geneLinks &amp;&amp; geneLinks.update('');
-                       }
-                     })
-                   });
-                }
-             }
-           });
-         },
-         onComplete : function () {
-           geneRefs.each(function(element) {
-             var geneLinks = element.next('div.gene-links');
-             if (geneLinks) {
-               geneLinks.removeClassName('loading');
-             }
-           });
-         }
-       }
-    );
+  var updateInfo = function(el, value, isSuggestSelected) {
+    var newValue = (!value || value == "") ? el.value : value;
```

Comment:
Javascript implements short-circuit evaulations of logical operators. That
means that if the left side is true in an `||`, the right side is not
evaluated. See
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_Operators#Short-Circuit_Evaluation

On Tue, Sep 15, 2015 at 9:55 AM Anton <notifications@github.com> wrote:

> In
> components/patient-data/ui/src/main/resources/PhenoTips/UIX_Field__genes.xml
> <https://github.com/phenotips/phenotips/pull/1873#discussion_r39512069>:
>
> > -                }
> > -             }
> > -           });
> > -         },
> > -         onComplete : function () {
> > -           geneRefs.each(function(element) {
> > -             var geneLinks = element.next('div.gene-links');
> > -             if (geneLinks) {
> > -               geneLinks.removeClassName('loading');
> > -             }
> > -           });
> > -         }
> > -       }
> > -    );
> > +  var updateInfo = function(el, value, isSuggestSelected) {
> > +    var newValue = (!value || value == "") ? el.value : value;
>
> I never understood how this works under the hood. In any other language, I
> think of || as a function taking 2 booleans. Clearly not the case with
> javascript.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/phenotips/phenotips/pull/1873/files#r39512069>.
>